contracts/core/script/helpverify/proxycreate3.go

```diff
@@ -73,9 +74,15 @@ func parseProxyCreate3Tx(ctx context.Context, chain ChainName, txHash common.Has
 
 	constructorArgs := creationCode[len(mustDecodeHex(bindings.TransparentUpgradeableProxyBin)):]
 
+	// implementation is first 20 byte word of constructor args
+	// TODO(kevin): EthAddress length != 32
+	impl, err := cast.EthAddress(constructorArgs[:20])
+	if err != nil {
+		return ProxyCreate3Tx{}, err
+	}
+
 	return ProxyCreate3Tx{
-		// implementation is first 32 byte word of constructor args
-		Implementation:  common.BytesToAddress(constructorArgs[:32]),
+		Implementation:  impl,
```

Comment:
`[:20]` won't work because it'll miss end of address.

packed address looks like this in constructor args
```
000000000000000000000000a863ae72ede2c6bd5158b3593d442a8163b427dd
```

BytesToAddress trims leading zeros

Reply:
refactored to `cast.EthAddress(constructorArgs[12:32])`